solr/core/src/java/org/apache/solr/search/neural/KnnQParser.java

```diff
@@ -84,30 +84,20 @@ public Query parse() {
   }
 
   private Query getFilterQuery() throws SolrException {
-    if (!isFilter()) {
+    boolean isSubQuery = recurseCount != 0;
+    if (!isFilter() && !isSubQuery) {
```

Comment:
Why is isFilter being checked?  Why is isSubQuery being checked?  In my experience, use of these is extremely rare.  isFilter... that would only be pertinent if the KNN query was specified in a filter query (`fq`), which is kind of surprising because AFAIK, KNN is for relevancy (scoring).  Does this query need to operate differently or to make more optimal decisions if it's in a filter query (to only filter and not rank)?